html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Application.pm

```diff
@@ -363,10 +367,15 @@ Render a template using Template Toolkit.
 sub _render {
     my ($self, $template, $args) = @_;
 
+    if(defined($args->{title})){
+        $self->title($args->{title});
+    }
+
     our $TT_OPTIONS = {
         AUTO_FILTER => 'html',
         RELATIVE => 1,
         INCLUDE_PATH => $self->profile->{_template_paths},
+        ENCODING => 'utf8',
```

Comment:
To make it easier to extend the template processor move the creation of the TT_OPTIONS into a method.